e2e/test/helpers/TestDeviceCallbackHandler.cs

```diff
@@ -52,33 +58,32 @@ public void Dispose()
             _receivedMessageCallbackSemaphore?.Dispose();
         }
 
-        internal async Task SetDeviceReceiveMethodAsync<T>(string methodName, object deviceResponse, T expectedServiceRequestJson)
+        internal async Task SetDeviceReceiveMethodAndRespondAsync<T,H>(H deviceResponsePayload)
         {
-            await _deviceClient.OpenAsync().ConfigureAwait(false);
             await _deviceClient.SetDirectMethodCallbackAsync(
                 (request) =>
                 {
+                    VerboseTestLogger.WriteLine($"{nameof(SetDeviceReceiveMethodAndRespondAsync)}: DeviceClient {_testDeviceId} callback method: {request.MethodName} with timeout {request.ResponseTimeout}.");
                     try
                     {
-                        VerboseTestLogger.WriteLine($"{nameof(SetDeviceReceiveMethodAsync)}: DeviceClient {_testDevice.Id} callback method: {request.MethodName} with timeout {request.ResponseTimeout}.");
-                        request.MethodName.Should().Be(methodName, "The expected method name should match what was sent from service");
+                        request.MethodName.Should().Be(ExpectedDirectMethodRequest.MethodName, "The expected method name should match what was sent from service");
+
+                        var expectedRequestPayload = (T)ExpectedDirectMethodRequest.Payload;
```

Comment:
We get the "expected" value from a class property which means that we can test different methods without having to reset the callback each time.